### PR TITLE
chore: CI support node18 (#3093)

### LINKED ISSUE #3093: feat: node 18
### Describe the feature you'd like to request

Node 18 is now LTS. Should probably make sure we support that.

### Describe the solution you'd like to see

Update CI to use Node 18

https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/.github/workflows/examples.yml#L39

https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/.github/workflows/lint.yml#L19

https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/.github/workflows/main.yml#L11

https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/.github/workflows/examples-old-nodes.yml#L11 (14->16 on this one? 👆👆)

https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/.github/workflows/examples-old-nodes.yml#L20

prob some other places too

### Desribe alternate solutions

idk

### Additional information

Renovate already updated our package json's engine to node 18: https://github.com/trpc/trpc/blob/ec301f0318a226a8c41796d0d00d11318d9f0d2c/package.json#L5

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by sun0day**
Closes #3093

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/next/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by sun0day** [comment]
Seems like `wait-on` doesn't work on  node18 when host is set `localhost`.

See https://github.com/jeffbski/wait-on/issues/109

We could change  `"test-start": "start-server-and-test start 3000 test:e2e"` to ` "test-start": "start-server-and-test start http://127.0.0.1:3000 test:e2e"` to bypass this problem for now.

File: examples/.interop/cloudflare-workers/src/client.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ global.fetch = fetch as any;
 const sleep = (ms = 100) => new Promise((resolve) => setTimeout(resolve, ms));
 
 async function main() {
-  const url = 'http://localhost:8787';
+  const url = 'http://127.0.0.1:8787';
```
Comment (sachinraja) [inline examples/.interop/cloudflare-workers/src/client.ts:11]: Why is this necessary?

**Review by KATT — DISMISSED** [review]
This is amazing. I will update the branch protection rules ASAP & merge this!

**Review by KATT — APPROVED** [review]
